docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,483 @@
+
+This method is passed both the :class:`~discord.Interaction` and a :class:`~re.Match` object, which contains the regular expression
+match details using the ``template`` attribute of the class as the pattern.
+
+In our case, we can cast the ``count`` group to an integer and pass it to the constructor.
+
+.. code-block:: python
+    :emphasize-lines: 12-15
+
+    class DynamicCounter(discord.ui.DynamicItem[discord.ui.Button], template=r'counter:(?P<count>\d+)'):
+        def __init__(self, count: int = 0) -> None:
+            self.count: int = count
+            super().__init__(
+                discord.ui.Button(
+                    label=f'Total: {count}',
+                    style=discord.ButtonStyle.primary,
+                    custom_id=f'counter:{count}',
+                )
+            )
+
+        @classmethod
+        async def from_custom_id(cls, interaction: discord.Interaction, match: re.Match[str], /):
+            count = int(match['count'])
+            return cls(count)
+
+
+And like all other interactions we'll need to define some kind of callback to handle the interaction.
+In this case we'll want to increment the counter and update the button label.
+
+Our callback might look like this:
+
+.. code-block:: python
+    :emphasize-lines: 17-21
+
+    class DynamicCounter(discord.ui.DynamicItem[discord.ui.Button], template=r'counter:(?P<count>\d+)'):
+        def __init__(self, count: int = 0) -> None:
+            self.count: int = count
+            super().__init__(
+                discord.ui.Button(
+                    label=f'Total: {count}',
+                    style=discord.ButtonStyle.primary,
+                    custom_id=f'counter:{count}',
+                )
+            )
+
+        @classmethod
+        async def from_custom_id(cls, interaction: discord.Interaction, item: discord.ui.Button, match: re.Match[str], /):
+            count = int(match['count'])
+            return cls(count)
+
+        async def callback(self, interaction: discord.Interaction) -> None:
+            self.count += 1
+            self.item.label = f'Total: {self.count}'
+            self.custom_id = f'counter:{self.count}'
+            await interaction.response.edit_message(view=self.view)
+
+
+Sending and Registering Dynamic Components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we've defined out dynamic component, we'll need to attach send a message which contains the component in a :class:`~discord.ui.View`.
+
+Let's quickly define a view which contains our dynamic component.
+
+.. code-block:: python
+
+    class CounterView(discord.ui.View):
+        def __init__(self) -> None:
+            super().__init__(timeout=None)
+            self.add_item(DynamicCounter())
+
+We'll also need some way to send a message with the view, so let's create a command to do that.
+
+.. code-block:: python
+    
+    @discord.app_commands.command()
```

Comment:
```suggestion
    @app_commands.command()
```

Exact same thing with the other suggestions in regards to `app_commands` package utilization